### DISCUSSION TITLE
Getting started with a small MinIO deployment...

### DISCUSSION THREAD

**Discussion by TheRealTomDavidson** [minio/minio#20669] [Q&A]
So my org needs to deploy a small local GitHub Enterprise server, but Actions requires S3 storage. Since for export-control reasons this all needs to be local, that means we need to set up a S3-compatible local store, and MinIO comes highly recommended. My IT team is not hugely Linux-savvy, however, and has some questions about the weight of the install. Do we actually need multiple nodes? How do we physically back up the box or VM on which it's running if MinIO demands "exclusive access" to the drives at all times? How does server recovery from backup work with a local MinIO instance?

**Comment by marktheunissen** [comment]
Hi @TheRealTomDavidson 

> Do we actually need multiple nodes?

If you don't need redundancy, recovery or high availability, then you can just use one server/node. For example in your Github actions, maybe all it's doing is generating ephemeral build artifacts and it doesn't matter if the MinIO drive crashes, you can just replace the drive and rebuild everything. 

Up to you to decide what level of service this needs to provide. For production critical deployments we always recommend HA.

> How do we physically back up the box or VM on which it's running if MinIO demands "exclusive access" to the drives at all times?

There are MinIO specific backup abilities, for example replication jobs can be enabled, or `mc mirror` can be executed to make a full copy of your data. 

Copying data from a running MinIO instance, by simple copying the file and directory structure underneath it, can result in corrupted objects. We don't recommend that. If you shut everything down first you could theoretically make a disk backup, however most people won't want to go this route because it requires downtime. The best option is to use the tools we provide.

> How does server recovery from backup work with a local MinIO instance?

Multi-node MinIO deployments are like RAID in a NAS, they have the concept of "backup" built in. If you lose a drive or multiple drives, you can rebuild the data by replacing the drive. 

Please reach out to sales@min.io for a more detailed look including support, licensing and setup advice.
